exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java

```diff
@@ -146,5 +163,77 @@ public OtlpGrpcMetricExporter build() {
     }
 
     private Builder() {}
+
+    /**
+     * Sets the configuration values from the given configuration map for only the available keys.
+     * This method looks for the following keys:
+     *
+     * <ul>
+     *   <li>{@code otel.otlp.metric.timeout}: to set the max waiting time for the collector to
+     *       process each metric batch.
+     * </ul>
+     *
+     * @param configMap {@link Map} holding the configuration values.
+     * @return this.
+     */
+    @Override
+    protected Builder fromConfigMap(
+        Map<String, String> configMap, NamingConvention namingConvention) {
+      configMap = namingConvention.normalize(configMap);
+      Long value = getLongProperty(KEY_METRIC_TIMEOUT, configMap);
+      if (value != null) {
+        this.setDeadlineMs(value);
+      }
+      return this;
+    }
+
+    /**
+     * Sets the configuration values from the given properties object for only the available keys.
+     * This method looks for the following keys:
+     *
+     * <ul>
+     *   <li>{@code otel.otlp.metric.timeout}: to set the max waiting time for the collector to
+     *       process each metric batch.
+     * </ul>
+     *
+     * @param properties {@link Properties} holding the configuration values.
+     * @return this.
+     */
+    @Override
+    public Builder readProperties(Properties properties) {
```

Comment:
Is the purpose of these overrides just to provide a handle for documenting the properties? If so, could we do that just in the class javadoc, rather than introducing a ton of lines of code simply for repeating documentation?

Reply:
As for now, all overrides have the function to provide better documentation w.r.t. the variables that they require.

Reply:
My preference would be to have the documentation not repeated, but have it just in the class-level javadoc and in the package-info.

Reply:
I agree! I would, however, suggest doing this doc refactoring in a follow-up PR where I address this in all the different classes that use a configuration.